### PR TITLE
initial cleanup of StudyResponsesMixin (move small general-purpose fn…

### PR DESCRIPTION
**Pull request by kimberscott**
…s to utils, move functions only used in one view, remove unused code); avoid using default argument values of []; try using .values() to reduce memory demands for frame data download function only

still trying to address #408

### CODE REVIEW

File: exp/utils.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            ]
+        else:
+            return [(key, value)]
+
+    items = [item for k, v in d.items() for item in expand(k, v)]
+
+    return dict(items)
+
+
+def csv_dict_output_and_writer(headerList):
```
Comment (Datamance) [inline exp/utils.py]: does black not change `headerList` to `header_list`? We may need to bring pyflakes as a dependency to make sure we adhere to PEP8

Comment (kimberscott) [reply exp/utils.py]: sorry about that - no, black does not change the variable names, I do run it!

Comment (Datamance) [reply exp/utils.py]: argh yeah ok after this PR I'll plug in a dependency and add an `invoke` script to run all the housekeeping goodies. I've been too reliant on my own PyCharm setup

Comment (kimberscott) [reply exp/utils.py]: ooh thanks! my apologies, camelcasing variables got drilled into me very early and has been hard to kick

**Comment by kimberscott** [comment]
will do!

**Pull request merged**